src/derivative_operators/BC_operators.jl

```diff
@@ -1,152 +1,173 @@
+        dx_l, dx_r = (dx[1:(order+nl-2)], reverse(dx[(end-order-nr+3):end]))
+        S_l = zeros(T, (nl-2, order+nl-2))
+        S_r = zeros(T, (nr-2, order+nr-2))
 
-Base.:*(Q::AffineBC, u::AbstractVector) = BoundaryPaddedVector(Q.a_l ⋅ u[1:length(Q.a_l)] + Q.b_l, Q.a_r ⋅ u[(end-length(Q.a_r)+1):end] + Q.b_r, u)
+        for i in 1:(nl-2)
+            S_l[i,:] = [transpose(calculate_weights(i, one(T), Array(one(T):convert(T, order+i)))) transpose(zeros(T, Int(nl-2-i)))]./(dx_l.^i)
+        end
 
-Base.size(Q::AbstractBC) = (Inf, Inf) #Is this nessecary?
-Base.length(Q::BoundaryPaddedVector) = length(Q.u) + 2
-Base.size(Q::BoundaryPaddedVector) = (length(Q),)
-Base.lastindex(Q::BoundaryPaddedVector) = Base.length(Q)
-
-function Base.getindex(Q::BoundaryPaddedVector,i)
-    if i == 1
-        return Q.l
-    elseif i == length(Q)
-        return Q.r
-    else
-        return Q.u[i-1]
-    end
-end
+        for i in 1:(nr-2)
+            S_r[i,:] = [transpose(calculate_weights(i, convert(T, order+i), Array(one(T):convert(T, order+i)))) transpose(zeros(T, Int(nr-2-i)))]./(dx_r.^i)
+        end
+        s0_l = S_l[:,1] ; Sl = S_l[:,2:end]
+        s0_r = S_r[:,end] ; Sr = S_r[:,(end-1):-1:1]
 
-function LinearAlgebra.Array(Q::AffineBC{T,V}, N::Int) where {T,V}
-    Q_L = [transpose(Q.a_l) transpose(zeros(T, N-length(Q.a_l))); Diagonal(ones(T,N)); transpose(zeros(T, N-length(Q.a_r))) transpose(Q.a_r)]
-    Q_b = [Q.b_l; zeros(T,N); Q.b_r]
-    return (Array(Q_L), Q_b)
-end
+        denoml = αl[2] .+ αl[3:end] ⋅ s0_l
+        denomr = αr[2] .+ αr[3:end] ⋅ s0_r
 
-function SparseArrays.SparseMatrixCSC(Q::AffineBC{T,V}, N::Int) where {T,V}
-    Q_L = [transpose(Q.a_l) transpose(zeros(T, N-length(Q.a_l))); Diagonal(ones(T,N)); transpose(zeros(T, N-length(Q.a_r))) transpose(Q.a_r)]
-    Q_b = [Q.b_l; zeros(T,N); Q.b_r]
-    return (Q_L, Q_b)
-end
+        a_l = -transpose(transpose(αl[3:end]) * Sl) ./denoml
+        a_r = -transpose(transpose(αr[3:end]) * Sr) ./denomr
 
-function SparseArrays.sparse(Q::AffineBC{T,V}, N::Int) where {T,V}
-    SparseMatrixCSC(Q,N)
+        b_l = -αl[1]/denoml
+        b_r = -αr[1]/denomr
+        new{T}(a_l,b_l,reverse!(a_r),b_r)
+    end
 end
 
-LinearAlgebra.Array(Q::PeriodicBC{T}, N::Int) where T = Array([transpose(zeros(T, N-1)) one(T); Diagonal(ones(T,N)); one(T) transpose(zeros(T, N-1))])
-SparseArrays.SparseMatrixCSC(Q::PeriodicBC{T}, N::Int) where T = [transpose(zeros(T, N-1)) one(T); Diagonal(ones(T,N)); one(T) transpose(zeros(T, N-1))]
-SparseArrays.sparse(Q::PeriodicBC{T}, N::Int) where T = SparseMatrixCSC(Q,N)
 
-function LinearAlgebra.Array(Q::BoundaryPaddedVector)
-    return [Q.l; Q.u; Q.r]
-end
 
-function Base.convert(::Type{Array},A::AbstractBC{T}) where T
-    Array(A)
-end
+#implement Neumann and Dirichlet as special cases of RobinBC
+NeumannBC(α::AbstractVector{T}, dx::Union{AbstractVector{T}, T}, order = 1) where T = RobinBC([zero(T), one(T), α[1]], [zero(T), one(T), α[2]], dx, order)
```

Comment:
No reason to allocate the vectors here. Should be a tuple. Let's mark this as a small improvement to make in a follow up  PR.